### PR TITLE
[component] Remove host.ReportFatalError

### PR DESCRIPTION
**Pull request by atoulme**
**Description:** 
 Remove `host.ReportFatalError`. It has been deprecated since 0.87.0.

**Link to tracking Issue:**
#6344

### CODE REVIEW

**Review by mx-psi — COMMENTED** [review]
Looks like there is an extension still using it

```
 # github.com/open-telemetry/opentelemetry-collector-contrib/extension/awsproxy [github.com/open-telemetry/opentelemetry-collector-contrib/extension/awsproxy.test]
Error: ./extension.go:29:9: host.ReportFatalError undefined (type component.Host has no field or method ReportFatalError)
```

File: .chloggen/remove_ReportFatalError.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []
```
Comment (mx-psi) [inline .chloggen/remove_ReportFatalError.yaml:25]:
```suggestion
change_logs: [api]
```

**Comment by atoulme** [comment]
I have opened https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31104 to fix the contrib tests.

**Comment by mx-psi** [comment]
```
# github.com/open-telemetry/opentelemetry-collector-contrib/extension/httpforwarder [github.com/open-telemetry/opentelemetry-collector-contrib/extension/httpforwarder.test]
?   	github.com/open-telemetry/opentelemetry-collector-contrib/extension/httpforwarder/internal/metadata	[no test files]
Error: ./extension.go:51:9: host.ReportFatalError undefined (type component.Host has no field or method ReportFatalError)
```

**Comment by atoulme** [comment]
> ```
> # github.com/open-telemetry/opentelemetry-collector-contrib/extension/httpforwarder [github.com/open-telemetry/opentelemetry-collector-contrib/extension/httpforwarder.test]
> ?   	github.com/open-telemetry/opentelemetry-collector-contrib/extension/httpforwarder/internal/metadata	[no test files]
> Error: ./extension.go:51:9: host.ReportFatalError undefined (type component.Host has no field or method ReportFatalError)
> ```

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31134 is open to fix this now.

**Comment by mx-psi** [comment]
Merged! Re-rolling the CI dice.... :game_die:

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@atoulme not yet fixed contrib.

**Comment by atoulme** [comment]
We're looking into it with Pablo, and there are a couple cases. One is tied up in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30693 and the other is the new otelarrow exporter, which reintroduced an occurrence. We're on it.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**